src/pyqasm/visitor.py

```diff
@@ -1990,8 +2002,38 @@ def _visit_function_call(
 
         return return_value, result
 
-    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> None:
-        pass
+    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> list[qasm3_ast.Statement]:
+        result = []
+
+        loop_counter = 0
+        max_iterations = self._loop_limit
+
+        while True:
+            cond_value = Qasm3ExprEvaluator.evaluate_expression(statement.while_condition)[0]
```

Comment:
As discussed in [another PR](https://github.com/qBraid/pyqasm/pull/199#pullrequestreview-2880776788), can you please add a check in here to check for classical registers in expression?

If there are any classical registers in expression which contain measurement results, we can't unroll the loop as we can not evaluate the condition at compile time.